### ISSUE TITLE
wofpy_config improvements

### ISSUE THREAD

**Issue by lsetiawan** [ODM2/WOFpy#139]
This is a follow up for #136.

`wofpy_config` is working as intended. I was able to execute 
- `--mode=development` and get only `odm2/timeseries` folder 
- `--mode=production` and get `odm2/timeseries` and `production_configs` folders

Though here's the catch, when I created `odm2/timeseries` with `development` mode the first time then using `production` mode the second time gave an error that `wofpyserver` has been created. 

```
Traceback (most recent call last):
  File "/home/lsetiawan/miniconda/envs/wofpy/bin/wofpy_config", line 11, in <module>
    sys.exit(main())
  File "/home/lsetiawan/miniconda/envs/wofpy/lib/python2.7/site-packages/wof/wofpy_config.py", line 67, in main
    makedirs(directory)
  File "/home/lsetiawan/miniconda/envs/wofpy/lib/python2.7/site-packages/wof/wofpy_config.py", line 45, in makedirs
    raise ValueError(msg(directory))
ValueError: Directory wofpyserver already exists.
```

@ocefpaf Have you thought of improving this soon in a way that I should get only `production_configs` when I run the second time? or what plans do you have? on those situations. Thanks.

**Comment by ocefpaf** [comment]
> hough here's the catch, when I created odm2/timeseries with development mode the first time then using production mode the second time gave an error that wofpyserver has been created.

That is by design. If the user is expected to change settings in those files we should not overwrite/add files them without a proper switch, like `--overwrite` for example. 

I can implement that. But, IMO, it is not too different from letting the user manage multiple calls using different directories, or explicitly removing the directory before calling the CLI again.

**Comment by emiliom** [comment]
@ocefpaf: @lsetiawan and I have discussed this. We also looked at your comments from the email you sent me today.

Here's what we want: ideally **two** overwrite options. We can call them "hard" vs "soft" overwrite:
1. **hard** overwrite: delete everything and replace
2. **soft** overwrite: do *not* delete anything that's already there, but add files that are not present in the target directory. This is the scenario @lsetiawan described when he opened this issue.

If your time is very limited and you can only focus on one of these, let's do the soft overwrite only, for now.

**Comment by ocefpaf** [comment]
> If your time is very limited and you can only focus on one of these, let's do the soft overwrite only, for now.

Time to implement that is not a problem. I can add both if that is what you want. However, I do advice against the extra complexity of two options. Users can easily manage multiple directories and/or deletions/renamings well. There is no need to cover all cases under the sun with options in CLI, in fact there is a risk of confounding users with too many options.

PS: I am implementing those today and you can decide on the PR.

**Comment by emiliom** [comment]
Got it, I see your concern. Since you're already implementing these, let's just leave them in there for now and prune back at a later time if we decide that's better.

**Comment by ocefpaf** [comment]
> Got it, I see your concern. Since you're already implementing these, let's just leave them in there for now and prune back at a later time if we decide that's better.

Also bare in mind that my "concerns" are only standards for general tools. If `wofpy` will live in the `ODM2` sphere only you can do pretty much anything you guys want to make it more convinient for your use case.

**Comment by emiliom** [comment]
> If wofpy will live in the ODM2 sphere only you can do pretty much anything you guys want to make it more convinient for your use case.

We do intend it to be used more widely. So, your advice is definitely appreciated!

**Comment by lsetiawan** [comment]
This was implemented in https://github.com/ODM2/WOFpy/pull/143. I've tested the functionality locally and everything works great! As expected. 

One thing to comment though, when I do `wofpy_config wofpyserver` then do it again, the folders shouldn't change and it didn't change, as expected. But, the CLI still prints `Adding files to directory wofpyserver` even though it's not adding anything new. Not a big deal, just an FYI. Thanks.
